packages/vaex-core/vaex/__init__.py

```diff
@@ -536,8 +541,7 @@ def iterator():
 
 
 def _from_csv_convert_and_read(filename_or_buffer, copy_index, maybe_convert_path, chunk_size, **kwargs):
-    # figure out the CSV file path
-    if isinstance(filename_or_buffer, str):
+    if isinstance(filename_or_buffer, str) & (maybe_convert_path is True):
```

Comment:
Maybe changing the order of checks would be a bit simpler to read, but LMK if you like it:
```
1. if `maybe_convert_path` is string => we know we have to use that as the file path
2. else if `filename_or_buffer` is a file path => use that
3. else fail.
```
Now the `maybe_convert_path is True` check works because we know that with the current code the `False` value would not be passed, but I'd rather not have to think about it.

Reply:
Sounds good to me. 
Please push the improvement :)

Reply:
Argh, next time I'm not going to make suggestion then 😛 . Ok will do.

Reply:
You can simply accept it Jovan

Reply:
Actually I'm not sure if I can update your PR. How can I?

Reply:
well I did not suggest it through a "Suggestion", so not sure if Jovan can. Let me try..

Reply:
Done.